airflow/kubernetes/pod_launcher.py

```diff
@@ -62,8 +64,20 @@ def __init__(self,
         self.extract_xcom = extract_xcom
 
     def run_pod_async(self, pod, **kwargs):
-        """Runs POD asynchronously"""
-        pod_mutation_hook(pod)
+        """Runs POD asynchronously
+
+        :param pod:
+        :type pod: k8s.V1Pod
+        """
+        try:
+            # attempts to run pod_mutation_hook using old pod, if this
+            # fails we attempt to run with k8s official pod
+            dummy_pod = convert_to_airflow_pod(pod)
+            pod_mutation_hook(dummy_pod)
+            dummy_pod = dummy_pod.to_v1_kubernetes_pod()
+            PodGenerator.reconcile_pods(pod, dummy_pod)
+        except AttributeError:
+            pod_mutation_hook(pod)
```

Comment:
I know this is for 1.10 but maybe we should consider adding a `PendingDeprecationWarning`?

Reply:
Yup creating a separate PR for that and adding all the old paths back with deprecation warning

Reply:
Created : https://github.com/apache/airflow/pull/10067